public/js/index.js

```diff
@@ -128,6 +128,17 @@ $(function () {
         console.error("Error saving config:", strError);
       }
     });
+
+  })
+
+  $('#updateSoftware').on('click', function () {
+    if (objSocket.connected) {
+      objSocket.emit('control_hat', '{“set_charger”: “ON”}')
+      objSocket.emit('update', `{"timestamp":"${Date.now()}", "version":"${RQ_PARAMS.UPDATE_VERSION}", "action":"UPDATE", "args":"UI"}`)
+    }else{
+      console.error('Socket is not connected to do the software update requested.')
```

Comment:
It will be more understandable for the student user if this says something more like:

Not connected to the robot so an UPDATE is not possible. Check the robot.

Reply:
done